startup error while parsing xrandr output.

I get the following exception when running any lwjgl application on this Lubuntu 14.04.02 Notebook.:

```
Exception in thread "LWJGL Application" java.lang.ExceptionInInitializerError
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setVSync(LwjglGraphics.java:446)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:118)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
	at org.lwjgl.opengl.LinuxDisplay.getAvailableDisplayModes(LinuxDisplay.java:954)
	at org.lwjgl.opengl.LinuxDisplay.init(LinuxDisplay.java:738)
	at org.lwjgl.opengl.Display.<clinit>(Display.java:138)
```
I traced the error to the xrandr parsing function but could not yet exactly pinpoint it.


Below is the output of xrandr. The offending lines are definitely the bottom ones with `LVDS-1-1 connected`. The laptop has 2 graphics card, I think this is probably the root cause: 
```
$> lspci | grep VGA
00:02.0 VGA compatible controller: Intel Corporation Mobile 4 Series Chipset Integrated Graphics Controller (rev 07)
01:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] RV710/M92 [Mobility Radeon HD 4330/4350/4550]
```
```
$> xrandr -q
Screen 0: minimum 320 x 200, current 1366 x 768, maximum 32767 x 32767
LVDS1 connected 1366x768+0+0 (normal left inverted right x axis y axis) 310mm x 174mm
   1366x768       60.0*+
   1360x768       59.8     60.0  
   1024x768       60.0  
   800x600        60.3     56.2  
   640x480        59.9  
VGA1 disconnected (normal left inverted right x axis y axis)
DP1 disconnected (normal left inverted right x axis y axis)
TV1 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
VGA-1-1 disconnected
LVDS-1-1 connected
   1366x768       60.0 +
   1280x720       59.9  
   1152x768       59.8  
   1024x768       59.9  
   800x600        59.9  
   848x480        59.7  
   720x480        59.7  
   640x480        59.4  
HDMI-1-0 disconnected
  1366x768 (0x45)   75.1MHz
        h: width  1366 start 1414 end 1446 total 1564 skew    0 clock   48.0KHz
        v: height  768 start  771 end  776 total  800           clock   60.0Hz
```

--- a/src/java/org/lwjgl/opengl/XRandR.java
+++ b/src/java/org/lwjgl/opengl/XRandR.java
@@ -92,15 +92,28 @@ public class XRandR {
 					}
 					name = sa[0];
 
-					// save position of this screen, will be used later when current modeline is parsed
-					if ( "primary".equals(sa[2]) ) {
-						parseScreenHeader(currentScreenPosition, sa[3]);
-						// save primary
-						primaryScreenIdentifier = name;
-					} else {
-						parseScreenHeader(currentScreenPosition, sa[2]);
-					}
-				} else {
+                                        // enabled?
+                                        if (sa.length > 2) {
+                                                // save position of this screen, will be used later when current modeline is parsed
+                                                if ( "primary".equals(sa[2]) ) {
+                                                        parseScreenHeader(currentScreenPosition, sa[3]);
+                                                        // save primary
+                                                        primaryScreenIdentifier = name;
+                                                } else {
+                                                        parseScreenHeader(currentScreenPosition, sa[2]);
+                                                }
+                                        } else {
+                                            // disabled screen, no position info available
+                                            currentScreenPosition[0] = 0;
+                                            currentScreenPosition[1] = 0;
+                                        }
+                                } else if ("disconnected".equals(sa[1])) {
+                                        // previous screen block is over
+                                        if (name != null) {
+                                                screens.put(name, possibles.toArray(new Screen[possibles.size()]));
+                                                name = null;
+                                        }
+                                } else if (name != null) { // only read modelines after a valid screen identifier
 					Matcher m = SCREEN_MODELINE_PATTERN.matcher(sa[0]);
 					if ( m.matches() ) {
 						// found a new mode line
@@ -113,7 +126,9 @@ public class XRandR {
 				}
 			}
 
-			screens.put(name, possibles.toArray(new Screen[possibles.size()]));
+                        if (name != null) {
+                                screens.put(name, possibles.toArray(new Screen[possibles.size()]));
+                        }
 
 			current = currentList.toArray(new Screen[currentList.size()]);
 

